### PR TITLE
TA Checkout Clear Queue

### LINKED ISSUE #378: As a TA, I want to have the option of clearing the queue when I check out
If I'm the last TA checking out before the office hour is scheduled to end, I want a prompt that lets me choose whether or not to clear the queue. If there is another scheduled office hour within 15 minutes after mine, I don't want to be prompted because questions should carry over.

### PR DESCRIPTION
**Pull request by dajinchu**
checkout

Closes #378

### CODE REVIEW

File: packages/server/test/util/factories.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export const CourseFactory = new Factory(CourseModel)
   .attr('icalURL', 'http://hi.com')
   .attr('enabled', true)
   .assocOne('semester', SemesterFactory)
-  .assocMany('officeHours', OfficeHourFactory);
+  .assocMany('officeHours', OfficeHourFactory, 0);
```
Comment (willstenzel) [inline packages/server/test/util/factories.ts:46]: Why is the 0 needed here? 

From my understanding this means no `officeHours` are being associated to a course. If that's the case could we remove the line all together?

File: packages/server/test/util/factories.ts [hunk 2]
```diff
@@ -59,7 +59,8 @@ export const QueueFactory = new Factory(QueueModel)
   .attr('room', 'Online')
   .assocOne('course', CourseFactory)
   .attr('allowQuestions', false)
-  .assocMany('officeHours', OfficeHourFactory);
+  .assocMany('officeHours', OfficeHourFactory)
+  .assocMany('staffList', UserFactory, 0);
```
Comment (willstenzel) [inline packages/server/test/util/factories.ts:63]: Minor thing, but same as above.. why is this needed?

File: packages/app/components/Today/TACheckinButton.tsx [hunk 3]
```diff
@@ -1,10 +1,12 @@
 import { API } from "@koh/api-client";
-import { Button, Input, Modal, Radio } from "antd";
+import { TACheckoutResponse } from "@koh/common";
```
Comment (willstenzel) [inline packages/app/components/Today/TACheckinButton.tsx:2]: Did you want to use this in the file?

File: packages/app/components/Today/TACheckinButton.tsx [hunk 4]
```diff
@@ -21,55 +23,119 @@ const CheckOutButton = styled(Button)`
...
 
+const CheckoutModalButtonRow = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+type CheckInButtonState =
+  | "CheckedIn"
+  | "CheckedOut"
+  | "CheckedIntoOtherQueueAlready";
```
Comment (willstenzel) [inline packages/app/components/Today/TACheckinButton.tsx:34]: `CheckedIntoOtherQueueAlready` is this supposed to be used?

File: packages/app/components/Today/TACheckinButton.tsx [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
-import { Button, Input, Modal, Radio } from "antd";
+import { TACheckoutResponse } from "@koh/common";
+import { Button, Modal, Space } from "antd";
+import moment from "moment";
 import { useRouter } from "next/router";
 import { ReactElement, useState } from "react";
 import styled from "styled-components";
 import { useCourse } from "../../hooks/useCourse";
-import { useProfile } from "../../hooks/useProfile";
+import { useQueue } from "../../hooks/useQueue";
```
Comment (willstenzel) [inline packages/app/components/Today/TACheckinButton.tsx:9]: Unused Import

**Review by willstenzel — APPROVED** [review]
Overall really solid PR. Found a few unused imports - maybe we could add that to the linter going forward

File: packages/server/test/util/factories.ts (hunk 1 above)
Comment (dajinchu) [reply packages/server/test/util/factories.ts:46]: it needs to make an empty array i believe

**Comment by willstenzel** [comment]
@dajinchu sorry for doing so much "linting" on this PR review. I'm not sure it's all that useful but I figured it would help keep the codebase a little cleaner overall. In the future should we add a rule to our eslint config so that unused imports get caught by the linter?

**Review by NEUDitao — APPROVED** [review]

File: packages/server/src/course/course.controller.ts [hunk 6]
```diff
@@ -101,24 +108,35 @@ export class CourseController {
...
+
+    // find out how long until next office hour
+    if (canClearQueue) {
+      const soon = moment().add(15, 'minutes').toDate();
+      const nextOfficeHour = await OfficeHourModel.findOne({
+        where: { startTime: MoreThanOrEqual(soon) },
+        order: {
+          startTime: 'ASC',
+        },
+      });
```
[the remark this answers is not in the document]
Comment (dajinchu) [reply packages/server/src/course/course.controller.ts:136]: queries where clauses are not the same. 1 sees if theres office hours in the next 15 minutes. 1 gets the next office hours after 15 minuutes
